cg/cli/backup.py

```diff
@@ -1,32 +1,110 @@
+    if db_flow_cell and db_flow_cell.has_backup:
+        return False, f"Flow cell: {flow_cell.id} is already backed-up"
+    if not flow_cell.is_flow_cell_ready():
+        return False, f"Flow cell: {flow_cell.id} is not ready"
+    if complete_file_path.exists():
+        return False, f"Encryption already completed for flow cell: {flow_cell.id}"
+    if pending_file_path.exists():
+        return False, f"Encryption already started for flow cell: {flow_cell.id}"
+    return True, None
+
+
 @click.group()
 @click.pass_obj
 def backup(context: CGConfig):
     """Backup utilities"""
     pass
 
 
+@backup.command("encrypt-flow-cells")
+@DRY_RUN
+@click.pass_obj
+def encrypt_flow_cells(context: CGConfig, dry_run: bool):
+    """Encrypt flow cells."""
+    status_db: Store = context.status_db
+    flow_cell_encryption_api = FlowCellEncryptionAPI(
+        binary_path=context.encryption.binary_path, config=context.dict(), dry_run=dry_run
+    )
+    flow_cell_encryption_api.slurm_api.set_dry_run(dry_run=dry_run)
+    encrypt_dir: Path = Path(context.backup.encrypt_dir)
+    flow_cells_dir = Path(context.flow_cells_dir)
+    LOG.debug(f"Search for flow cells ready to encrypt in {flow_cells_dir}")
+    for flow_cell_dir in flow_cells_dir.iterdir():
+        if not flow_cell_dir.is_dir():
+            continue
+        LOG.debug(f"Found directory: {flow_cell_dir}")
+        try:
+            flow_cell = FlowCellDirectoryData(flow_cell_path=flow_cell_dir)
+        except FlowCellError:
+            continue
+        db_flow_cell: Optional[Flowcell] = status_db.get_flow_cell_by_name(
+            flow_cell_name=flow_cell.id
+        )
+        flow_cell_encrypt_dir = Path(encrypt_dir, flow_cell.full_name)
+        flow_cell_encrypt_file_path_prefix = Path(flow_cell_encrypt_dir, flow_cell.id)
+        complete_file_path: Path = flow_cell_encrypt_file_path_prefix.with_suffix(
+            FileExtensions.COMPLETE
+        )
+        pending_file_path: Path = flow_cell_encrypt_file_path_prefix.with_suffix(
+            FileExtensions.PENDING
+        )
+        is_requirement_meet, error_msg = _is_encryption_possible(
+            db_flow_cell=db_flow_cell,
+            flow_cell=flow_cell,
+            complete_file_path=complete_file_path,
+            pending_file_path=pending_file_path,
+        )
+        if not is_requirement_meet:
+            LOG.debug(error_msg)
+            continue
+        flow_cell_encrypt_dir.mkdir(exist_ok=True, parents=True)
+        flow_cell_encryption_api.create_pending_file(pending_path=pending_file_path)
+        flow_cell_encryption_api.encrypt_flow_cell(
+            flow_cell_dir=flow_cell_dir,
+            flow_cell_id=flow_cell.id,
+            flow_cell_encrypt_dir=flow_cell_encrypt_dir,
+            flow_cell_encrypt_file_path_prefix=flow_cell_encrypt_file_path_prefix,
+            pending_file_path=pending_file_path,
+            complete_file_path=complete_file_path,
+        )
```

Comment:
I would wrap these functions in the `FlowCellEncryptionAPI` into a function called `start_encryption` to hide the logic from the CLI command

Reply:
Fixed